src/main/java/core/basesyntax/service/impl/FruitTransactionServiceImpl.java

```diff
@@ -0,0 +1,45 @@
+package core.basesyntax.service.impl;
+
+import static core.basesyntax.model.FruitTransaction.Operation;
+
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.FruitTransactionService;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class FruitTransactionServiceImpl implements FruitTransactionService {
+    @Override
+    public List<FruitTransaction> getFruitTransactions(List<String> transactions) {
+        if (transactions == null) {
+            throw new RuntimeException("Transactions cannot be null");
+        }
+        transactions.forEach(transaction -> {
+            if (transaction == null) {
+                throw new RuntimeException("Transaction cannot be null");
+            }
+        });
```

Comment:
lets move this check in createNewFruitTransaction method(so we wont go through all transactions twice)

Reply:
Done